python_modules/dagster/dagster/_model/__init__.py

```diff
@@ -15,3 +17,9 @@ def __init__(self, **data: Any) -> None:
     class Config:
         extra = "forbid"
         frozen = True
+
+    def model_copy(self, *, update: Optional[Dict[str, Any]] = None, deep: bool = False) -> Self:
+        if pydantic.__version__ >= "2":
+            return super().model_copy(update=update, deep=deep)  # type: ignore
+        else:
+            return super().copy(update=update, deep=deep)
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: lines 22 to 26]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->weird does our typechecking environment use pydantic v1?<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/20736?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
that does seem to be the case: https://github.com/dagster-io/dagster/blob/master/pyright/master/requirements-pinned.txt#L421